README.md

```diff
@@ -157,136 +139,121 @@ symbols, it will fail at class definition time. This program will throw:
 ```js
 protocol I { a; b; }
 
-class C implements I {
-  [I.a]() {}
-  // note the missing implementation of I.b
-}
-```
-
-### minimal implementations
-
-Minimal implementations can be expressed using protocol inheritance.
-
-```js
-// Applicative elided
-protocol Monad extends Applicative {
-  bind;
-  join;
-  kleisli() {}
-}
-
-// two possible minimal implementations for Monad
-protocol MonadViaBind extends Monad {
-  [Monad.join]() { /* default implementation in terms of bind elided */ }
-}
-protocol MonadViaJoin extends Monad {
-  [Monad.bind]() { /* default implementation in terms of join elided */ }
-}
-
-class C implements MonadViaBind {
-  [Monad.bind]() {}
-}
-class D implements MonadViaJoin {
-  [Monad.join]() {}
+class C {
+  implements I {
+    a() {}
+    // b() {}
+    // ^__ throws PartialImplError
+  }
 }
 ```
 
-### `Protocol.implement`
+### `implFor <class>`
 
 An important aspect of this proposal is that it needs to be possible to apply
 a protocol to an existing class.
 
 ```js
 protocol Functor {
   map;
-}
 
-Array.prototype[Functor.map] = Array.prototype.map;
-Protocol.implement(Array, Functor);
+  implFor Array {
+    map() { return this.map.apply(this, arguments) }
+  }
+}
 ```
 
-`Protocol.implement` accepts zero or more protocols following the class.
+Having `implFor` allows extending existing prototypes for one's own purposes
+while preventing multiple conflicting implementations using the same protocol
+symbols. With this format, there's only two ways to implement a protocol:
 
-```js
-protocol I {}
-protocol K {}
+1. Using the new `implements` keyword in `class` bodies, which allows new class definitions to use a pre-existing protocol.
+2. Using `implFor` to make an existing class comply with the protocol.
```

Comment:
Can you elaborate on how this is enforced?

Reply:
You literally cannot define a protocol with this current proposal unless you go one of these two routes, and two protocols can't clobber each other... So there's literally no way to multi-define them.

Reply:
And it’s impossinle for me to install them later? How is this enforced without freezing a bunch of things?